File: pascals-triangle/pascals-triangle.py
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        ans = []
        for i in range(1,numRows+1):
            temp = [1]
            m = 1
            for j in range(1,i):
                m*=(i-j)
                m//=j
                temp.append(m)
            ans.append(temp)
        return ans

File: 118-pascals-triangle/pascals-triangle.java
class Solution {
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> ans = new ArrayList<>();
        for (int i = 0; i < numRows; i++) {
            List<Integer> list = new ArrayList<>();
            for (int j = 0; j < i + 1; j++) {
                if (j == 0 || j == i)
                    list.add(1);
                else {
                    List<Integer> prev = ans.get(i - 1);
                    list.add(prev.get(j - 1) + prev.get(j));
                }
            }
            ans.add(list);
        }
        return ans;
    }
}